src/util/misc.js

```diff
@@ -290,8 +290,147 @@
+     * - target is transformed by object but is unaware of it.
+     * - if target is positioned at (0,0) it is positioned at object's center.
+     *
+     * `sibling` relation means target exists in the same coordinate plane as the object we relate to.
+     * In other words they both relate to the same (0,0), agree on every point and are drawn in the same transformed plane.
+     *
+     * @static
+     * @memberOf fabric.util
+     * @see {fabric.util.transformPointRelativeToCanvas} for transforming relative to canvas
+     * @param {fabric.Object} [sourceObject] object that point currently relates to
+     * @param {fabric.Object} destinationObject object that returned point should relate to
+     * @param {'sibling'|'child'} [relationToSource] optional if `sourceObject` is `null`
+     * @param {'sibling'|'child'} relationToDestination
+     * @returns {fabric.Point} transformed point
+     */
+    calcTransformationBetweenObjectPlanes: function (
+      sourceObject, destinationObject,
+      relationToSource, relationToDestination
+    ) {
+      var from = sourceObject ?
+            fabric.util.getTransformMatrixByObject(sourceObject, relationToSource) :
+            fabric.iMatrix.concat(),
+          to = fabric.util.getTransformMatrixByObject(destinationObject, relationToDestination);
+      //  actually we are looking for the transformation between the destination plane to the source plane
+      //  because the object will exist on the destination plane and we want it to seem unchanged by it we reverse the destination matrix
+      //  this is de facto a linear mapping (which can help explain why the order is reversed if the explanation didn't)
+      //  think of how the target will be transformed once it's on the destination plane
+      //  first it reverses the effect of the plane and then it transforms itself with the source transformation, achieving what we wanted
+      return fabric.util.multiplyTransformMatrices(fabric.util.invertTransform(to), from);
+    },
+
+    /**
+     * Sends a point from the source coordinate plane to the destination coordinate plane.\
+     * From the canvas/viewer's perspective the point remains unchanged.
+     *
+     * `child` relation means `point` exists in the coordinate plane created by the object we relate to.
+     * In other words, point is measured acoording to object's center point
+     * meaning that if `point` is equal to (0,0) it is positioned at object's center.
+     *
+     * `sibling` relation means `point` exists in the same coordinate plane as the object we relate to.
+     * In other words they both relate to the same (0,0) and agree on every point.
+     *
+     * @static
+     * @memberOf fabric.util
+     * @see {fabric.util.transformPointRelativeToCanvas} for transforming relative to canvas
+     * @param {fabric.Point} point
+     * @param {fabric.Object} sourceObject object that point currently relates to
```

Comment:
in calcTransformationBetweenObjectPlanes sourceObject is optional. So it could be here too?

Reply:
done